lib/manager/pub/readme.md

```diff
@@ -0,0 +1,277 @@
+Example from <https://github.com/brianegan/flutter_architecture_samples/blob/master/example/firestore_redux/pubspec.yaml>:
+
+```yaml
+environment:
+  sdk: '>=2.0.0 <3.0.0'
+
+dependencies:
+  meta: '>=1.1.0 <2.0.0'
+  redux: ^3.0.0
+  flutter_redux: ^0.5.0
+  flutter:
+    sdk: flutter
+  flutter_architecture_samples:
+    path: ../../
+  firebase_flutter_repository:
+    path: ../firebase_flutter_repository
+
+dev_dependencies:
+  test: ^1.3.0
+  mockito: ^3.0.0
+  flutter_driver:
+    sdk: flutter
+  flutter_test:
+    sdk: flutter
+  integration_tests:
+    path: ../integration_tests
+  todos_repository_flutter:
+    path: ../todos_repository_flutter
+```
+
+---
+
+#### List all the sources/syntaxes of dependencies that can be extracted:
+
+- [**SDK**](https://www.dartlang.org/tools/pub/dependencies#sdk)
+  ```yaml
+  dependencies:
+    flutter_driver:
+      sdk: flutter
+      version: ^0.0.1
+  ```
+
+- [**Version constraints**](https://www.dartlang.org/tools/pub/dependencies#version-constraints)
+  ```yaml
+  dependencies:
+    meta: '>=1.1.0 <2.0.0'
+    flutter_redux: 0.5.0
+  ```
+
+- [**Caret syntax**](https://www.dartlang.org/tools/pub/dependencies#caret-syntax)
+  ```yaml
+  dependencies:
+    redux: ^3.0.0
+  ```
+
+- [**Git packages**](https://www.dartlang.org/tools/pub/dependencies#git-packages)
+  ```yaml
+  dependencies:
+    kittens:
+      git: git://github.com/munificent/kittens.git
+  ```
+  ```yaml
+  dependencies:
+    kittens:
+      git: git@github.com:munificent/kittens.git
+  ```
+  ```yaml
+  dependencies:
+    kittens:
+      git:
+        url: git://github.com/munificent/kittens.git
+        ref: some-branch
+  ```
+  ```yaml
+  dependencies:
+    kittens:
+      git:
+        url: git://github.com/munificent/cats.git
+        path: path/to/kittens
+  ```
+
+- [**Path packages**](https://www.dartlang.org/tools/pub/dependencies#path-packages)
+  ```yaml
+  dependencies:
+    transmogrify:
+      path: /Users/me/transmogrify
+  ```
+
+- [**Hosted packages**](https://www.dartlang.org/tools/pub/dependencies#hosted-packages)
+  ```yaml
+  dependencies:
+    transmogrify:
+      hosted:
+        name: transmogrify
+        url: http://your-package-server.com
+      version: ^1.4.0
+  ```
+
+---
+
+#### Describe which types of dependencies above are supported and which will be implemented in future:
+
+All
+
+## Versioning
+
+#### What versioning scheme do the package files use?
+
+<https://www.dartlang.org/tools/pub/dependencies>
+
+---
+
+#### Does this versioning scheme support range constraints, e.g. `^1.0.0` or `1.x`?
+
+Yes
+
+---
+
+#### Is this package manager used for applications, libraries, or both? If both, is there a way to tell which is which?
+
+Both. No.
+
+---
+
+#### If ranges are supported, are there any cases when Renovate should pin ranges to exact versions if rangeStrategy=auto?
+
+Dev dependencies can always be pinned.
+
+## Lookup
+
+#### Is a new datasource required? Provide details
+
+Not sure.
```

Comment:
We'll need to support the pub API as a new datasource. Example URL: https://pub.dartlang.org/api/packages/url_launcher

I haven't found a "spec" for it yet so we may just need to implement by example.